plugins/trezor/plugin.py

```diff
@@ -26,6 +26,9 @@ class TrezorCompatibleKeyStore(Hardware_KeyStore):
     def get_derivation(self):
         return self.derivation
 
+    def is_segwit(self):
+        return self.derivation.startswith("m/49'/")
```

Comment:
I don't think this is a good idea (is there a wallet property for SegWit that we can use?)

Reply:
inddeed, you should use keystore.is_segwit()
it should not be a property of the plugin, because there is a single plugin object for many wallets

Reply:
this looks ok, it is a property of the KeyStore

Reply:
@ecdsa It would be nice to be able to use a different derivation for whatever reason.

Reply:
@saleemrashid that is a consequence of the bip39+bip43 design, which was imposed by the trezor development team. I believe it is a bad standard, but we are forced to follow it in the trezor plugin.

Reply:
@ecdsa It's not a consequence AFAIK because I could just change that to say any other derivation and it will still work. We should keep track of whether to use SegWit through a different means.

Reply:
@saleemrashid yes you can, but we do not want to expose users to a system where they have to remember more than their seed. (at least not with electrum seeds)

Reply:
@ecdsa But they wouldn't, because this "SegWit property" would be set by Electrum.

Reply:
@saleemrashid how would electrum know about this property?

Reply:
@ecdsa Well, the TREZOR plugin would know when the wallet is imported (it has to choose between `m/44'` and `m/49'` anyway)

Reply:
@saleemrashid so you're proposing to add a checkbox in the trezor restore dialog, where the user enters the derivation? that would work for me

Reply:
@ecdsa We already allow the user to enter the derivation. I'm proposing to allow them to choose between SegWit and non-SegWit for any derivation.

Reply:
@saleemrashid I am not opposed to that. I will let @jhoenicke decide if he wants to add it.